trace-sdk/src/androidTest/java/io/bitrise/trace/network/MetricSenderIntegrationTest.java

```diff
@@ -0,0 +1,116 @@
+package io.bitrise.trace.network;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+import androidx.test.filters.LargeTest;
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import io.bitrise.trace.InstrumentedTestRequirements;
+import io.bitrise.trace.configuration.ConfigurationManager;
+import io.bitrise.trace.data.collector.cpu.ApplicationCpuUsageDataCollector;
+import io.bitrise.trace.data.collector.cpu.SystemCpuUsageDataCollector;
+import io.bitrise.trace.data.collector.memory.ApplicationUsedMemoryDataCollector;
+import io.bitrise.trace.data.collector.memory.SystemMemoryDataCollector;
+import io.bitrise.trace.data.collector.view.ApplicationStartUpDataListener;
+import io.bitrise.trace.data.dto.ApplicationStartData;
+import io.bitrise.trace.data.dto.ApplicationStartType;
+import io.bitrise.trace.data.dto.Data;
+import io.bitrise.trace.data.dto.FormattedData;
+import io.bitrise.trace.data.management.formatter.cpu.ApplicationCpuDataFormatter;
+import io.bitrise.trace.data.management.formatter.cpu.SystemCpuDataFormatter;
+import io.bitrise.trace.data.management.formatter.memory.ApplicationUsedMemoryDataFormatter;
+import io.bitrise.trace.data.management.formatter.memory.SystemMemoryDataFormatter;
+import io.bitrise.trace.data.management.formatter.view.ApplicationStartUpDataFormatter;
+import io.bitrise.trace.session.ApplicationSessionManager;
+import io.bitrise.trace.BuildConfig;
+import io.bitrise.trace.test.DataTestUtils;
+import io.opencensus.proto.metrics.v1.Metric;
+import retrofit2.Response;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Integration tests that send a metric to the backend and expect confirmation by headers.
+ */
+@LargeTest
+public class MetricSenderIntegrationTest {
+
+    @Before
+    public void setUp() {
+        ConfigurationManager.getDebugInstance(BuildConfig.TRACE_TOKEN);
```

Comment:
nit: as it is initialised always with the same key, it would be enough to init/reset once per test class, no need to do it before each test case.